packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -272,6 +273,40 @@ export class PluginTree extends TreeImpl {
         }, update);
     }
 
+    override markAsChecked(node: Mutable<TreeNode>, checked: boolean): void {
+        function findParentsToChange(child: TreeNode, nodes: TreeNode[]): void {
+            if (child.parent) {
+                if (child.parent.checkboxInfo !== undefined && child.parent.checkboxInfo.checked !== checked) {
+                    if (!checked || child.parent.children.every(candidate => candidate.checkboxInfo?.checked || candidate.checkboxInfo === undefined || candidate === child)) {
+                        nodes.push(child.parent);
+                        findParentsToChange(child.parent, nodes);
+                    }
+                }
+            }
+        }
+
+        function findChildrenToChange(parent: TreeNode, nodes: TreeNode[]): void {
+            if (CompositeTreeNode.is(parent)) {
+                parent.children.forEach(child => {
+                    if (child.checkboxInfo !== undefined && child.checkboxInfo.checked !== checked) {
```

Comment:
Minor:
```suggestion
                    if (child.checkboxInfo?.checked !== checked) {
```

Reply:
Right, nevermind. I missunderstood the reason why the `child.checkboxInfo !== undefined` exists in the first place. Nothing to see here... :)